File: frameworks/java/quarkus/UnitTests/src/test/java/unittests/MiddlewareTests.java
package unittests;

import static unittests.Http.get;

import io.quarkus.test.junit.QuarkusTest;
import org.junit.jupiter.api.Tag;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.ValueSource;

@QuarkusTest
class MiddlewareTests {

    // rb:test middleware.none,middleware.four,middleware.sixteen
    @ParameterizedTest
    @Tag("middleware.none")
    @Tag("middleware.four")
    @Tag("middleware.sixteen")
    @ValueSource(strings = {"none", "four", "sixteen"})
    void everyLayerPassesTheRequestThrough(String layers) {
        Answer.is(Expected.json("items.small.json"), get("/middleware/" + layers));
    }
}

File: frameworks/java/ktor/UnitTests/src/test/kotlin/unittests/MiddlewareTests.kt
package unittests

import io.ktor.client.request.get
import kotlin.test.assertEquals
import org.junit.jupiter.api.Tag
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.ValueSource

class MiddlewareTests {
    // rb:test middleware.none,middleware.four,middleware.sixteen
    @ParameterizedTest
    @Tag("middleware.none")
    @Tag("middleware.four")
    @Tag("middleware.sixteen")
    @ValueSource(strings = ["none", "four", "sixteen"])
    fun `the layers in front of the handler leave the answer alone`(layers: String) = corpusTest {
        assertEquals(Expected.json("items.small.json"), client.get("/middleware/$layers").json())
    }
}
